src/clr-angular/popover/tooltip/tooltip.spec.ts

```diff
@@ -35,3 +20,27 @@ describe('Tooltip component', function() {
     `,
 })
 class SimpleTest {}
+
+export default function(): void {
+  describe('Tooltip component', function() {
+    let context: TestContext<ClrTooltip, SimpleTest>;
+
+    addHelpers();
+    beforeEach(function() {
+      context = this.create(ClrTooltip, SimpleTest);
+    });
+
+    describe('TypeScript API', () => {
```

Comment:
What about a test case where I set an `id` manually, check does it get pulled through

Reply:
Added an input for id that is used to associate content id <---> trigger aria-describedby
If a custom string is passed thats what is used. If the id value is set on the input with {undefined|null|''} an empty string is used, and if no id is set at all clarity handles setting a unique id for the tooltip complex.